src/hb-ot-layout.cc

```diff
@@ -1116,6 +1116,143 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
   return false;
 }
 
+/**
+ * hb_ot_layout_get_feature_name_ids:
+ * @face: #hb_face_t to work upon
+ * @feature_tag: ssXX and cvXX tag
+ * @table_tag:
+ * @feature_index:
+ * @variations_index:
+ * @label_id: (out) (allow-none): The ‘name’ table name ID that specifies a string
+ *            for a user-interface label for this feature. (May be NULL.)
+ * @tooltip_id: (out) (allow-none): The ‘name’ table name ID that specifies a string
+ *              that an application can use for tooltip text for this
+ *              feature. (May be NULL.)
+ * @sample_id: (out) (allow-none): The ‘name’ table name ID that specifies sample text
+ *             that illustrates the effect of this feature. (May be NULL.)
+ * @num_named_parameters: (out) (allow-none):  Number of named parameters. (May be zero.)
+ * @first_param_id: (out) (allow-none): The first ‘name’ table name ID used to specify
+ *                  strings for user-interface labels for the feature
+ *                  parameters. (Must be zero if numParameters is zero.)
+ *
+ * Return value: true if could find any feature with the tag, false otherwise
+ *
+ * Since: REPLACEME
+ **/
+hb_bool_t
+hb_ot_layout_get_feature_name_ids (hb_face_t      *face,
+				   hb_tag_t        feature_tag,
+				   hb_tag_t        table_tag,
+				   unsigned int    feature_index,
+				   unsigned int    variations_index,
+				   unsigned int   *label_id,             /* OUT.  May be NULL */
+				   unsigned int   *tooltip_id,           /* OUT.  May be NULL */
+				   unsigned int   *sample_id,            /* OUT.  May be NULL */
+				   unsigned int   *num_named_parameters, /* OUT.  May be NULL */
+				   unsigned int   *first_param_id        /* OUT.  May be NULL */)
+{
+  static_assert ((OT::FeatureVariations::NOT_FOUND_INDEX == HB_OT_LAYOUT_NO_VARIATIONS_INDEX), "");
+  const OT::GSUBGPOS &g = get_gsubgpos_table (face, table_tag);
+
+  const OT::Feature &f = g.get_feature_variation (feature_index, variations_index);
+
+  const OT::FeatureParams &feature_params = f.get_feature_params ();
+  if (&feature_params != &Null (OT::FeatureParams))
```

Comment:
I know there is a better way for this but I used `size` as the base for this initially which goes for the same way.